src/server/main_service.cc

```diff
@@ -1549,19 +1549,22 @@ optional<bool> StartMultiEval(DbIndex dbid, CmdArgList keys, ScriptMgr::ScriptPa
   return false;
 }
 
-static std::string FullAclCommandFromArgs(CmdArgList args) {
+static std::string FullAclCommandFromArgs(CmdArgList args, std::string_view name) {
   ToUpper(&args[1]);
   // Guranteed SSO no dynamic allocations here
-  return std::string("ACL ") + std::string(args[1].begin(), args[1].end());
+
+  return absl::StrCat(name, " ", std::string(args[1].begin(), args[1].end()));
 }
 
 std::pair<const CommandId*, CmdArgList> Service::FindCmd(CmdArgList args) const {
   const std::string_view command = facade::ToSV(args[0]);
-  if (command == "ACL") {
+  std::string_view acl = "ACL";
+  acl = registry_.RenamedOrOriginal(acl);
+  if (command == acl) {
     if (args.size() == 1) {
       return {registry_.Find(ArgS(args, 0)), args};
     }
-    return {registry_.Find(FullAclCommandFromArgs(args)), args.subspan(2)};
+    return {registry_.Find(FullAclCommandFromArgs(args, acl)), args.subspan(2)};
```

Comment:
Same here. We could check that the command has some flag like `two-part` and then try looking for that command + args[1]. It would automatically add support for commands like `XGROUP` or `XINFO`.... I don't think it's needed now, it's just that you just provided a quick fix for ACL stuff

Reply:
>  command has some flag like two-part 

That's what I had in mind and that's why I wrote we can do the checks for subcommands early on. Basically, at the constructor of `CommandId` we can add a boolean `is_subcommand` and each time we create a command with `space` and two words we know that it's a subcommand and we set the boolean to true. That way, we can check if a command is a subcommand by `cid.IsSubcommand()`.

Reply:
Yes, and that way we don't need RenamedOrOriginal and other workarounds